### ISSUE TITLE
RFE: fullscreen single 3DS screen and a shortcut to switch it.

### ISSUE THREAD

**Issue by i30817** [citra-emu/citra#1552]
I think the ability to put in the main screen fullscreen and switch screens with a controller or keyboard button (or even a mouse gesture) would be pretty nice on some games that only use the secondary screen for non-essential functions like maps or inventories.

Another idea would be to send the images to different screens available.

Yet another idea would be to resize and overlap the smaller screen on the fullscreen main 3ds one, only that is far too game specific probably (size, best location etc), so it's probably not worth it.

**Comment by ErdeFB** [comment]
Lack of such feature has been an annoyance for me in some DS emulators. +1

**Comment by JayFoxRox** [comment]
This is closely related to other frontend enhancements such as #139 and #204.

I have also thought about doing this but don't have the time for it.
However, this is one of the possibilities I came up with and which I'd have implemented.

+1 for bunneis approach of turning every screen into it's own window.
But I'd add a theater mode to that so you can fill everything around the 3DS screen in a specific color on the host screen(s). Think of this like what lightbox does. There'd be a configurable list where you can set which host screens will be affected by theater mode.
Fullscreen resolution would automaticly be that of the host screen (or configurable later if people complain?) and the host screen for fullscreen would be the one the guest window is placed on at the time of the switch.

```
Configuration dialog:

  --- Screens

      Arrangement:  <Top Screen in front, Bottom screen in back,
                     Bottom Screen in front, Top screen in back>
               (maybe by making the following boxes draggable) // Hotkey: Tab (configurable as toggle or hold; hold as default)

      --- Top screen:
        Visible       [x] // Hotkey: F3
        Rotation      <0, 90, 180, 270> // Hotkey (Spin): Ctrl F3 (Shift is modifier for spin backwards)
        Opacity:      <Slider, default 100%>

      --- Bottom screen
        Visible       [x] // Hotkey: F4
        Rotation      <0, 90, 180, 270> // Hotkey (Spin): Ctrl F4 (Shift is modifier for spin backwards)
        Opacity:      <Slider, default 100%>

  --- Theater mode Mask:

     (List:)
       [ ] Display 1
       [ ] Display 2
       [ ] Display N
       [x] Active Display (Top-Screen) // Hotkey: None
       [x] Active Display (Bottom-Screen) // Hotkey: None

     Color: [dark grey default, color selection button]

---

Menu + Context menu:

  [ ] Fullscreen // Hotkey: F10
  [ ] Theater mode // Hotkey: F11
  ```

**Comment by i30817** [comment]
Tywald linked suggestion of making the secondary screen resizable, draggable and invisible and turn the opacity to 100% when moused over would make the 3rd option here more usable.

**Comment by BynariStar** [comment]
Seems to be fixed by #1753 and #3001

**Comment by JayFoxRox** [comment]
Agreed, closed by #3001
